src/python/pants/base/specs.py

```diff
@@ -388,6 +388,35 @@ def resolved_files(self) -> Tuple[str, ...]:
         return self._snapshot.files
 
 
+@dataclass(frozen=True)
+class FilesystemMergedSpec(FilesystemResolvedSpec):
+    """When multiple file arguments are given that resolve to the same owning target, this type
+    allows grouping the arguments into a single FilesystemSpec."""
+
+    original_globs: Tuple[str, ...]
+    merged_files: Tuple[str, ...]
```

Comment:
This is confusing- you're not merging file content, you're merging lists of files.  I think just calling this `files` is fine. 

Ditto, you can just call `original_globs` `globs`. 

Basically, neither the naming here nor the docstring needs to be influenced by "how an instance of this came into being". Once it exists, it just has globs and files. Does that make sense?